R/run-nmtran.R

```diff
@@ -0,0 +1,375 @@
+
+  run_dir <- as.character(fs::path_real(dir))
+  nmtran_input <- mod_path_new <- file.path(run_dir, basename(mod_path))
+
+  # Preprocess with nmtran_presort
+  if(!is.null(nmtran_presort_exe)){
+    # NM-TRAN input is now output from nmtran_presort
+    nmtran_input <- file.path(
+      paste0(fs::path_ext_remove(mod_path_new), "_presort.", fs::path_ext(mod_path_new))
+    )
+
+    presort.p <- processx::run(
+      command = nmtran_presort_exe,
+      stdin = mod_path_new,
+      stdout = nmtran_input,
+      wd = run_dir,
+      error_on_status = TRUE
+    )
+  }
+
+  # Run NM-TRAN
+  nmtran.p <- processx::run(
+    command = nmtran_exe,
+    args = cmd_args,
+    stdin = nmtran_input,
+    stdout = "|",
+    wd = run_dir,
+    stderr_to_stdout = TRUE,
+    error_on_status = FALSE
+  )
+
+  # Assign status
+  status_val <- nmtran.p$status
+  if(is.na(status_val)){
+    cli::cli_abort("NM-TRAN terminated unexpectedly")
+  }else if(status_val == 0){
+    status <- "Finished Running"
+  }else{
+    status <- "Failed. See errors."
+  }
+
+  # Tabulate NM-TRAN results
+  nmtran_results <- list(
+    nmtran_model = basename(nmtran_input),
+    run_dir = run_dir,
+    status = status,
+    status_val = status_val,
+    output = nmtran.p$stdout
+  )
+
+  return(nmtran_results)
+}
+
+
+#' Helper for finding and reading in a `bbi.yaml` file given a `bbi_nonmem_model`
+#' object or explicit `.config_path`.
+#' @inheritParams run_nmtran
+#' @noRd
+get_bbi_yaml_path <- function(.mod = NULL, .config_path = NULL){
+  if(!is.null(.mod)){
+    check_model_object(.mod, "bbi_nonmem_model")
+    model_dir <- get_model_working_directory(.mod)
+  }
+  bbi_yaml_path <- .config_path %||% file.path(model_dir, "bbi.yaml")
+
+  if(!file_exists(bbi_yaml_path)){
+    if(is.null(.config_path)){
+      cli::cli_abort(
+        c(
+          "No bbi configuration was found in the execution directory.",
+          "i" = "Please run {.fun bbr::bbi_init} with the appropriate directory to continue."
+        )
+      )
+    }else{
+      cli::cli_abort("No bbi configuration was found at {.path {(.config_path)}}")
+    }
+  }
+
+  return(bbi_yaml_path)
+}
+
+
+#' Parse `.bbi_args` and return the three expected `nmfe_options` for `NM-TRAN`
+#'  in the correct format
+#' @inheritParams run_nmtran
+#' @param nmfe_options named list of nmfe options defined in a `bbi.yaml` file.
+#'
+#' @details
+#'
+#' Combines NONMEM submission args and consolidates to `NMFE` arguments only
+#'  - The arguments of interest are `prdefault`, `tprdefault`, and `maxlim`,
+#'  which impact the evaluation of `NM-TRAN`
+#'  - Priority: `.bbi_args` > model yaml > `bbi.yaml`
+#'
+#' @examples
+#'
+#' \dontrun{
+#'
+#'  bbi_yaml_path <- get_bbi_yaml_path(.mod)
+#'  bbi_yaml <- yaml::read_yaml(bbi_yaml_path)
+#'
+#'  parse_nmtran_args(
+#'    .mod,
+#'    .bbi_args = list(nm_version = "nm74gf", prdefault = TRUE),
+#'    nmfe_options = bbi_yaml$nmfe_options
+#'  )
+#'
+#' }
```

Comment:
[ okay as is ]

This are developer docs for an un-exported function. An example that doesn't run is likely to grow stale and incorrect. If this function's usage within this file isn't sufficient to be an example for developers to look at, I'd rather see a test of this function that shows its usage and gives the developer something easier to grab onto.